presto-main/src/main/java/com/facebook/presto/resourcemanager/ResourceManagerClusterStateProvider.java

```diff
@@ -141,6 +141,7 @@ public ResourceManagerClusterStateProvider(
             for (Map.Entry<String, InternalNodeState> nodeEntry : ImmutableList.copyOf(nodeStatuses.entrySet())) {
                 if ((System.currentTimeMillis() - nodeEntry.getValue().getLastHeartbeatInMillis()) > nodeStatusTimeout.toMillis()) {
                     nodeStatuses.remove(nodeEntry.getKey());
+                    resourceGroupStates.remove(nodeEntry.getKey());
```

Comment:
Should we have this expiry based on its own internal lastHeartbeatTime?

Reply:
Done